pkg/store/promproxy.go

```diff
@@ -249,7 +249,7 @@ func encodeChunk(ss []prompb.Sample, mint int64) (storepb.Chunk_Encoding, []byte
 
 // translateAndExtendLabels transforms a metrics into a protobuf label set. It additionally
 // attaches the given labels to it, overwriting existing ones on colllision.
-func translateAndExtendLabels(m []prompb.Label, extend labels.Labels) []storepb.Label {
+func (p *PrometheusProxy) translateAndExtendLabels(m []prompb.Label, extend labels.Labels) []storepb.Label {
```

Comment:
Why this and above? no need for that if these function are not using anything from this struct.

Reply:
OK, can see now, these methods needs to be different for TSDB Store and you want to stick to the same naming.. Lgtm

Reply:
Yea, it's just about namespacing them.